### ISSUE TITLE
Dependency on Kotlin standard library

### ISSUE THREAD

**Issue by Philipp91** [mikepenz/Android-Iconics#466]
I've updated to `4.0.0` (AndroidX version `1.0.2`, gradle build tools `28.0.3`, Android Studio `3.4.2`) and updated some call sites like `sizeRes(X)` to `size(IconicsSize.res(X))`. However, that didn't compile because there is a second (unused) overload `size(() -> IconicsSize?)` that gets compiled to a function that accepts `kotlin.jvm.functions.Function0` and this class wasn't available on my class path (I have a plain Java project), so it failed to resolve the right overload.

The (right?) solution is to add a dependency on `org.jetbrains.kotlin:kotlin-stdlib`, so currently that's `implementation org.jetbrains.kotlin:kotlin-stdlib:1.3.31`.

Ideally, the Android-Iconics would tell Gradle about this dependency, so that users don't have to do anything. Alternatively, there could be a remark in the migration guide and the main setup guide (under "1. Provide the gradle dependency") that tells users that they might need this dependency.

Edit: Just found it in the GitHub release notes. Probably worth adding to the other two places as well, as those are discoverable through GitHub search. Also there it mentions a dependency on `androidx.core:core-ktx`, so for non-Kotlin projects that would be `androidx.core:core`?

**Comment by mikepenz** [comment]
The dependency is in the `pom.xml`: https://bintray.com/mikepenz/maven/download_file?file_path=com%2Fmikepenz%2Ficonics-core%2F4.0.0%2Ficonics-core-4.0.0.pom

So the dependency will also be loaded by gradle automatically. 

Is there anything wrong in your setup which might prevent this automatic transitive resolving of the dependency?

How did you specify the lib in your gradle?

**Comment by Philipp91** [comment]
I do specify `@aar` (e.g. `implementation 'com.mikepenz:iconics-core:4.0.0@aar'`), which I had to add at some point to make it work. And [apparently](https://stackoverflow.com/a/30157801/636372) that excludes transitive dependencies by default. Judging from your readme file, `@aar` is only needed for the typeface dependencies, but not (anymore?) for the main library. So I dropped it there, and was able to remove the `kotlin-stdlib` dependency again. I guess the other alternative would be `{ transitive = true }`. Thank you, sorry for the noise!
